app/queries/patient_summary_query.rb

```diff
@@ -0,0 +1,32 @@
+class PatientSummaryQuery
+  def self.call(*args)
+    new(*args).call
```

Comment:
I've found this `.call` abstraction to be a bit unnecessary, assuming this is here only because it allows us to:
1. make neat looking scopes: [reference](http://craftingruby.com/posts/2015/06/29/query-objects-through-scopes.html). 
2. do things like `PatientSummaryQuery.where(...)`. This works only if we want to use the default parameters though, which isn't often the case.

My thoughts on 1: This query object is a presentation level concern (filtering etc.) and will never be/shouldn't be used in the `PatientSummary` model. Providing for it to be used as a scope seems a bit circular. Wdyt?

Reply:
I've seen this pattern used _outside_ the context you've referenced, @prabhanshuguptagit . The crux being: we should operate with an instance of the `PatientSummaryQuery` to leverage advantages of OOP. This class method is then created as a convenience method so that callers don't have to worry about instantiating this object themselves. They can say "Hey, class, take care of this", while maintaining an OO implementation under the hood.

I quite like this.